CodeStyleExercise/ColorsViewController.swift

```diff
@@ -8,88 +8,62 @@
 
 import UIKit
 
-class ColorsViewController : UIViewController, UICollectionViewDataSource, UITextFieldDelegate
-{
-
-    class constants {
-        static let DFLTNUMCOLORS:Int=12
-    }
-
-    @IBOutlet weak var collectionView:UICollectionView!
-    @IBOutlet weak var colorCountTextField: UITextField!
-
+final class ColorsViewController: UIViewController, UICollectionViewDataSource, UITextFieldDelegate {
+    
+    @IBOutlet private weak var collectionView: UICollectionView!
+    @IBOutlet private weak var colorCountTextField: UITextField!
+    
     var ct: Int? = 12
-
+    
+    class Constants {
+        static let defaultColorNumber = 12
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.collectionView.register(UINib(nibName: "ColorsCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "Cell")
-
-        self.colorCountTextField.placeholder = "12"
+        collectionView.register(UINib(nibName: "ColorsCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "Cell")
```

Comment:
Instead of hardcoding in the nib name and reuse identifier strings, store them as a constant. That way, if they change, only one string needs to be changed rather than a lot of them. For reuse identifiers, since they're tied to the cell they identify, they can be stored as a static constant in the cell itself. See the solution code for an example of this.